stream.go

```diff
@@ -205,7 +205,9 @@ func (cn *conn) StreamQuery(q string, quit chan struct{}) (msgs chan *XLogDataMs
 					err := binary.Read(buf, binary.BigEndian, &(msg.Header))
 					ERROR(err, "message read failed")
 
-					msg.Data = []byte((*r)[24:])
+					data := (*r)[24:]
```

Comment:
we can reuse this buffer space after some time (after p2q sent the message) so maybe putting it in a pool + some manipulations may be worth.
I want to avoid it if possible, as i dont see the need for it.

Reply:
@schattian do u think it is possible to copy it in p2q - not here? for example here https://github.com/remerge/p2q/blob/master/service.go#L220-L222

The reason is - our knowledge of pq is very limited - and p2q is much safer place to do changes...

WDYT?

Reply:
pq's stream.go (this file) is just from this fork and not from upstream, so i think we should maintain that.
I think putting the logic in p2q is just leaking the unsafety of this, and i dont see why that's better, it's kinda confusing

But yep, in the end it's the same for our purpose